sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.students.controller.getStudentProfile.e2e-spec.ts

```diff
@@ -0,0 +1,88 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getProviderInstanceForModule,
+  getStudentToken,
+} from "../../../../testHelpers";
+import { saveFakeStudent } from "@sims/test-utils";
+import { determinePDStatus, getUserFullName } from "../../../../utilities";
+import { UserService } from "../../../../services";
+import { TestingModule } from "@nestjs/testing";
+import { AuthModule } from "../../../../auth/auth.module";
+import { IdentityProviders } from "@sims/sims-db";
+
+describe("StudentInstitutionsController(e2e)-searchStudents", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let appModule: TestingModule;
+  const endpoint = "/students/student";
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    appModule = module;
+  });
+
+  it("Should get the student profile when student requests.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(appDataSource);
+
+    // Mock user service for auth module
+    const userService = await getProviderInstanceForModule<UserService>(
+      appModule,
+      AuthModule,
+      UserService,
+    );
+    userService.getUserLoginInfo = jest.fn(() =>
+      Promise.resolve({
+        id: student.user.id,
+        isActive: true,
+        studentId: student.id,
+        identityProviderType: IdentityProviders.BCSC,
+      }),
+    );
+    // Get any student user token
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        firstName: student.user.firstName,
+        lastName: student.user.lastName,
+        fullName: getUserFullName(student.user),
+        email: student.user.email,
+        gender: student.gender,
+        dateOfBirth: student.birthDate,
+        contact: {
+          address: {
+            addressLine1: student.contactInfo.address.addressLine1,
+            provinceState: student.contactInfo.address.provinceState,
+            country: student.contactInfo.address.country,
+            city: student.contactInfo.address.city,
+            postalCode: student.contactInfo.address.postalCode,
+            canadaPostalCode: student.contactInfo.address.postalCode,
+            selectedCountry: student.contactInfo.address.selectedCountry,
+          },
+          phone: student.contactInfo.phone,
+        },
+        pdStatus: determinePDStatus(student),
+        validSin: student.sinValidation.isValidSIN,
+        sin: student.sinValidation.sin,
```

Comment:
Not part of the PR but the SIN is not supposed to be exposed for the student. Apparently it "leaked" to the student client once the SIN was enabled on the Ministry side.
It would be very important to have this test adjusted to ensure that it will not happen in the future again.
Can we please have a small ticket and a small PR to have it fixed right away?
By the way, thanks for introducing this test that brought this to light.

Reply:
Created a ticket: [1910](https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/gh/bcgov/sims/1910).